modules/doc/content/ncrc/applications/ncrc_develop.md.template

```diff
@@ -0,0 +1,147 @@
+you through an update process when performed multiple times).
+
+Proper development of {{ApplicationName}} requires a compiler, the {{bison}} repository, and
+additional support libraries. The following instructions aim at preparing your machine or
+environment for such development.
+
+If you encounter any errors during any of these steps, you need to stop, and figure out a solution
+before continuing. Please see our [Troubleshooting](help/troubleshooting.md) guide, which also
+contains information on how to reach us in the event troubleshooting fails.
+
+## Compiler Stack id=compiler
+
+If you plan on using your personal machine we recommend using our Conda installation method. If
+you are instead operating on one of our [!ac](INL) [!ac](HPC) clusters, you need only load a couple
+of modules.
+
+- If +Personal Machine+: Follow the [Conda Installation](installation/install_miniconda.md) instructions and then perform the following:
+
+  ```bash
+  mamba create -n moose moose-tools moose-libmesh
+  ```
+
+- If +[!ac](INL) [!ac](HPC) Sawtooth or Lemhi+ (required each time you log in):
+
+  ```bash
+  module load use.moose moose-dev
+  ```
+
+For those operating on other [!ac](HPC) clusters, you should contact that system's administrator,
+and use their recommended compiler stack. It will need to satisfy to our
+[minimum requirements](sqa/minimum_requirements.md).
+
+!alert warning title=Conda packages may be suboptimal on HPC
+Conda packages have the advantage of being able to work on most machines, but on [!ac](HPC)
+clusters, this flexibility potentially comes at the cost of the optimizations developed by
+[!ac](HPC) administrators.
+
+## Cloning {{ApplicationName}} id=cloning
+
+Instructions for obtaining a proper clone of {{ApplicationName}} can be found at
+[NCRC/{{ApplicationName}}](ncrc/applications/ncrc_level4_{{ApplicationLower}}.md).
+
+## Building {{ApplicationName}} id=build
+
+Building {{ApplicationName}} begins with applying upstream changes, updating your submodules and/or
+Conda packages, and finally running `make`. Before we begin, +remember these two important
+warnings+:
+
+!alert warning title=You updated Conda?
+`Then you should also perform a submodule update`.
+
+!alert warning title=You performed a submodule update?
+`Then you should also perform a Conda update or rebuild PETSc/libMesh`.
+
+The symmetry importance comes from our Conda packages and support libraries being version controlled
+by and within the MOOSE repository. Updates to MOOSE (the `{{binary}}/moose` submodule) requires you
+to stay on top of any Conda updates (if Personal Machine), or rebuilding PETSc and libMesh (if using
+[!ac](HPC) resources).
+
+### Applying Upstream Changes
+
+If this is your first time following these instructions there should be no updates. However, we will
+perform an update to familiarize yourself with the entire process.
+
+{{ApplicationName}} developers will expect users to have performed the following sections, whenever
```

Comment:
I think this sentence can be deleted. If you want to emphasize that this WHOLE section needs to be completed to update, maybe you can make an `!alert note` after this box with your sentence "Note this step does not actually modify...".